sdk/identity/src/token_credentials/default_credentials.rs

```diff
@@ -1,27 +1,32 @@
+use std::{sync::Arc, time::Duration};
+
 use crate::{
-    timeout::TimeoutExt,
-    token_credentials::cache::TokenCache,
-    {AzureCliCredential, ImdsManagedIdentityCredential},
+    timeout::TimeoutExt, token_credentials::cache::TokenCache, AppServiceManagedIdentityCredential,
+    AzureCliCredential, EnvironmentCredential, TokenCredentialOptions,
+    VirtualMachineManagedIdentityCredential,
 };
 use azure_core::{
     auth::{AccessToken, TokenCredential},
     error::{Error, ErrorKind, ResultExt},
 };
-use std::time::Duration;
 
-#[derive(Debug)]
 /// Provides a mechanism of selectively disabling credentials used for a `DefaultAzureCredential` instance
 pub struct DefaultAzureCredentialBuilder {
+    options: TokenCredentialOptions,
     include_environment_credential: bool,
-    include_managed_identity_credential: bool,
+    include_app_service_managed_identity_credential: bool,
+    include_virtual_machine_managed_identity_credential: bool,
     include_azure_cli_credential: bool,
 }
 
 impl Default for DefaultAzureCredentialBuilder {
     fn default() -> Self {
         Self {
+            options: TokenCredentialOptions::default(),
             include_environment_credential: true,
-            include_managed_identity_credential: true,
+            include_app_service_managed_identity_credential: true,
+            // Unable to quickly detect if running in Azure VM, so it is disabled by default.
+            include_virtual_machine_managed_identity_credential: false,
```

Comment:
During the Azure Identity v-team, this came up. 

I believe the stated stance was thus:
1. The v-team acknowledged the potential frustration that IMDS authentication includes a delay due to the network connection
2. Authenticating generally happens up-front, only once, and then auth cached & refreshed as needed.  As such, the additional delay at initialization was minimal and acceptable.
3. Someone on the v-team was investigating if there was a faster way to tell if the app was running in a VM, which if found, all of the SDKs could built upon in the future.
4. The "Default Azure Credential" should therefore have this enabled by default.  This ensures continuity across languages.

Reply:
I think it is a bug to enable it by default in the other Azure SDKs based on the open issues. We do not need to add this bug.

Reply:
I put it back.